### ISSUE TITLE
Crash on starlette middleware in mount endpoints (StaticFiles)

### ISSUE THREAD

**Issue by Syniex** [elastic/apm-agent-python#1131]
**Describe the bug**: Elastic middleware crash on access to static files

**Environment (please complete the following information)**
- OS: [e.g. Linux]
- Python version: 3.9.4
- Framework and version [e.g. Django 2.1]: FastAPI
- APM Server version: 7.12.1
- Agent version: 7.12.1

```python
from __future__ import annotations

import uvicorn
from elasticapm.contrib.starlette import ElasticAPM, make_apm_client
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles

app = FastAPI(routes=[])
apm_config = {
    "SERVICE_NAME": "testing",
    "ENABLED": True,
    "SERVER_URL": "http://localhost:8200",
    "CAPTURE_HEADERS": True,
    "CAPTURE_BODY": "all",
}
apm = make_apm_client(apm_config)
app.add_middleware(ElasticAPM, client=apm)
app.mount("/static",StaticFiles(directory="/"),name="static")


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", debug=False)
```

The issue is happens when elastic running the function _get_route_name.
the middleware will try to loop over the routes where starlette will return None in some cases:
```python
  File "/home/syniex/.local/lib/python3.9/site-packages/elasticapm/contrib/starlette/__init__.py", line 233, in _get_route_name
    for route in routes:
                 └ None

TypeError: 'NoneType' object is not iterable

```
Starlette code that makes the issue:
```python
@property
    def routes(self) -> typing.List[BaseRoute]:
        return getattr(self.app, "routes", None)
```

**Comment by Syniex** [comment]
[Starlette#1182](https://github.com/encode/starlette/issues/1182#issue-890379365)

**Comment by acnebs** [comment]
Seems like this was introduced in this commit: https://github.com/elastic/apm-agent-python/commit/a5e66603254bb9b126172a233b50be5cb88d9949#diff-e15c015370e75a124090c9f0ca4d6f3e015f87b2b0b2f2f1081efd0622ab448e

**Comment by Sparkycz** [comment]
https://github.com/elastic/apm-agent-python/blob/master/elasticapm/contrib/starlette/__init__.py#L238
`if isinstance(route, Mount):` -> `if isinstance(route, Mount) and route.routes:` and add tests.

**Comment by nourwolf** [comment]
I get the same error when mounting a prometheus's ASGI app to `/metrics` endpoint. The app has no routes as it is a simple function that returns some payload.

**Comment by bjoern-meier-by** [comment]
Hi, we just hit this issue as well when mounting static files for e.g. the swagger UI and would be interested if the work on the the open PR is still ongoing as the apm client now breaks our app and we did not find a workaround so far.

**Comment by bvader** [comment]
Hiya, @Sparkycz  Any Updates on this? I have a customer eagerly awaiting this as well...the PR seems stalled?

**Comment by basepi** [comment]
The PR is waiting on me. We have some upstream test failures caused by the new version of `aiopg`, which I'm working to fix here: https://github.com/elastic/apm-agent-python/pull/1178

Once that's merged we should be able to get #1137 passing and merged. Thanks for your patience.
